lib/rust/ensogl/core/src/debug/stats.rs

```diff
@@ -15,48 +15,137 @@
 
 use enso_prelude::*;
 
+use enso_web::Performance;
 use js_sys::ArrayBuffer;
 use js_sys::WebAssembly::Memory;
 use wasm_bindgen::JsCast;
 
 
 
+// ====================
+// === TimeProvider ===
+// ====================
+
+/// An entity that can retrieve current time.
+pub trait TimeProvider {
+    /// Returns current time, measured in milliseconds.
+    fn now(&self) -> f64;
+}
+
+impl TimeProvider for Performance {
+    fn now(&self) -> f64 {
+        self.now()
+    }
+}
+
+
+
 // =============
 // === Stats ===
 // =============
 
-/// Structure containing all the gathered stats.
-#[derive(Debug, Clone, CloneRef)]
-pub struct Stats {
-    rc: Rc<RefCell<StatsData>>,
+/// Tracks stats related to the current rendering frame, providing methods for modifying and
+/// retrieving their values.
+pub type Stats = StatsOverTime<Performance>;
+
+
+// === StatsOverTime ===
+
+/// Tracks stats related to the current rendering frame, providing methods for modifying and
+/// retrieving their values.
+#[derive(Debug, CloneRef)]
+pub struct StatsOverTime<T: TimeProvider> {
+    rc: Rc<RefCell<StatsCollector<T>>>,
```

Comment:
I don't understand why a "stats collector" which is put in Rc<RefCell<...>> is named `StatsOverTime`. So a shared stats collector are stats over time?

Reply:
Alternatives `SharedStatsHandler`. (Or `Shared`...whatever the inner struct is).